### DISCUSSION TITLE
More dummy methods about visibility

### DISCUSSION THREAD

**Discussion by okuramasafumi** [mruby/mruby#5458] [Ideas]
Hi, I recently noticed that as described in https://github.com/mruby/mruby/blob/master/doc/limitations.md mruby has no method visibility implemented. As a result, methods such as `public` are "dummy" methods.

My proposal is to have more dummy methods like them. For example, `private_instance_methods` is not implemented. If it works as an alias of `instance_methods`, it becomes a bit easier to copy CRuby code to mruby.

**Comment by matz** [comment]
I am not sure about making `private_instance_methods` an alias to `instance_methods` is a good idea when we don't provide method visibility. The result from `private_instance_methods` may better be empty since there are no `private` methods in `mruby`. But the empty result from the method can be useless at the same time.

**Reply by okuramasafumi** [reply]
Thanks, as a CRuby user, I'd be surprised to get empty result when I call `private_instance_methods` unless I see some warnings.
